README.md

```diff
@@ -1,12 +1,267 @@
+# Syndesis
+
+Syndesis is a single page application built with React.
+
+## Table of Contents
+
+* [Architecture](#architecture)
+  * [syndesis](#syndesis-1)
+  * [packages](#packages)
+      * [api](#packagesapi)
+      * [models](#packagesmodels)
+      * [ui](#packagesui)
+      * [utils](#packagesutils)
+      * [syndesis-context](#packagessyndesis-context)
+  * [typings](#typings)
+* [Installation](#installation)
+* [Building](#building)
+* [Scripts](#scripts)
+* [Roadmap](#roadmap)
+
+## Architecture 
+
+We use [Lerna](https://github.com/lerna/lerna) to streamline the development process; the most common operations, 
+like building the project or running the development mode can be done directly from the project root.  
+
+Code is split in many packages, organized as a monorepo using [Yarn's workspaces](https://yarnpkg.com/lang/en/docs/workspaces/). 
+The workspace is configured like this:
+
 ```
+"syndesis",
+"packages/*"
+"typings/*"
+```
+
+### syndesis
+
+`syndesis` is the main application. It handles the authentication against Syndesis's OAuth Server, and provides the main
+ app layout where "sub-apps" can be injected. 
+
+![](doc/assets/syndesis-chrome.png)
+
+It also provides an [API for sub-apps](#packagessyndesis-context) - in the form of a [React's Context](https://reactjs.org/docs/context.html) - for 
+interact with it, eg. closing the navigation bar, or redirecting to another sub-app.
+
+It's built with [create-react-app](https://github.com/facebook/create-react-app).
+
+#### Development server
+
+```bash
+# From the repository root
+$ yarn watch:app
+```
+
+Or
+
+```bash
+# From the syndesis folder
+$ yarn start
+```
+
+
+#### First time setup
```

Comment:
Might be good to move this section before the "Development Server" section, so the steps are more sequential.